scripts/zkai_attention_d128_value_adapter_gate.py

```
@@ -0,0 +1,882 @@
+            post_stat = os.fstat(fd)
+            if (pre_stat.st_dev, pre_stat.st_ino, pre_stat.st_size) != (
+                post_stat.st_dev,
+                post_stat.st_ino,
+                post_stat.st_size,
+            ):
+                raise AttentionD128ValueAdapterError(f"source path changed while reading: {path}")
+            if not stat_module.S_ISREG(post_stat.st_mode):
+                raise AttentionD128ValueAdapterError(f"source path must remain a regular file: {path}")
+            raw = os.read(fd, MAX_SOURCE_BYTES + 1)
+            if len(raw) > MAX_SOURCE_BYTES:
+                raise AttentionD128ValueAdapterError(f"source path exceeds size limit after open: {path}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)